egs/wsj/s5/steps/tensorflow/vanilla_rnnlm.py

```diff
@@ -0,0 +1,319 @@
+# Copyright 2015 The TensorFlow Authors. All Rights Reserved.
+#           2017 Hainan Xu, adapted to be used in Kaldi for lattice rescoring
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+# ==============================================================================
+
```

Comment:
There should be a comment (or doc-string, maybe) somewhere near the top that explains what this script does and how it's supposed to be invoked.